### PR TITLE
Add driver for bh1745 ambient light sensor

### PR DESCRIPTION
**Pull request by gblach**

### CODE REVIEW

File: drivers/sensor/bh1745/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+#  Copyright (c) 2019 Nordic Semiconductor ASA
+#  Copyright (c) 2022 Grzegorz Blach
+#
+#  SPDX-License-Identifier: Apache-2.0
+#
+
+zephyr_library()
+zephyr_library_sources(bh1745.c)
+zephyr_library_sources_ifdef(CONFIG_BH1745_TRIGGER bh1745_trigger.c)
+
```
Comment (MaureenHelm) [inline drivers/sensor/bh1745/CMakeLists.txt:11]: Please remove the extra blank line

File: drivers/sensor/bh1745/bh1745.c [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+	}
+
+	if (unlikely(!data->ready)) {
+		LOG_INF("Device is not initialized yet");
+		return -EBUSY;
+	}
+
+	LOG_DBG("Fetching sample...\n");
+
+	if (i2c_reg_read_byte(data->i2c, DT_REG_ADDR(DT_DRV_INST(0)),
```
Comment (MaureenHelm) [inline drivers/sensor/bh1745/bh1745.c:62]: Please use the dt variant for all i2c transactions (`i2c_reg_read_byte_dt` and friends)

File: drivers/sensor/bh1745/bh1745.c [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+	.attr_set = &bh1745_attr_set,
+	.trigger_set = &bh1745_trigger_set,
+#endif
+};
+
+static struct bh1745_data bh1745_data;
+
+DEVICE_DEFINE(bh1745, DT_INST_LABEL(0),
+	      bh1745_init, NULL, &bh1745_data, NULL,
+	      POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY, &bh1745_driver_api);
```
Comment (MaureenHelm) [inline drivers/sensor/bh1745/bh1745.c:312]: Please update this driver to support multiple instances and use `DT_INST_FOREACH_STATUS_OKAY`

File: drivers/sensor/bh1745/bh1745_trigger.c [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+/* Callback for active sense pin from BH1745 */
+static void bh1745_gpio_callback(const struct device *dev,
+				 struct gpio_callback *cb,
+				 uint32_t pins)
+{
+	struct bh1745_data *drv_data =
+		CONTAINER_OF(cb, struct bh1745_data, gpio_cb);
+
+	gpio_pin_interrupt_configure(dev, DT_INST_GPIO_PIN(0, int_gpios),
```
Comment (MaureenHelm) [inline drivers/sensor/bh1745/bh1745_trigger.c:29]: Please use the dt variant for all gpio transactions (`gpio_pin_interrupt_configure_dt` and friends)

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thank you for the contribution!

Please add this sensor to `tests/drivers/build_all/sensor/` to ensure it builds in CI.

File: drivers/sensor/bh1745/CMakeLists.txt (hunk 1 above)
Comment (gblach) [reply drivers/sensor/bh1745/CMakeLists.txt]: Done

File: drivers/sensor/bh1745/bh1745.c (hunk 2 above)
Comment (gblach) [reply drivers/sensor/bh1745/bh1745.c]: Done

File: drivers/sensor/bh1745/bh1745.c (hunk 3 above)
Comment (gblach) [reply drivers/sensor/bh1745/bh1745.c]: Done

File: drivers/sensor/bh1745/bh1745_trigger.c (hunk 4 above)
Comment (gblach) [reply drivers/sensor/bh1745/bh1745_trigger.c]: Done

**Pull request closed without merging**